File: Swea/2301/16/이경훈_1209_Sum.py
for t in range(10):
  tc = int(input())
  data = [list(map(int, input().split())) for _ in range(100)]
  
  max = 0
  
  for r in range(100):
    sum = 0
    for c in range(100):
      sum += data[r][c]
    if sum > max:
      max = sum

  for c in range(100):
    sum = 0
    for r in range(100):
      sum += data[r][c]
    if sum > max:
      max = sum
      
  sum1 = 0
  sum2 = 0
  
  for r in range(100):
    sum1 += data[r][r]  
  if sum1 > max:
    max = sum1

  for r in range(100):
    sum2 += data[r][99-r]  
  if sum2 > max:
    max = sum2
    
  print(f'#{tc} {max}')

File: Swea/2301/16/이경훈_1209_Sum.java
import java.io.*;
import java.util.*;

public class 이경훈_1209_Sum {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int[][] data = new int[100][100];
		for (int tc = 0; tc < 10; tc++) {
			int t = Integer.parseInt(br.readLine());

			for (int i = 0; i < 100; i++) {
				StringTokenizer st = new StringTokenizer(br.readLine());
				for (int j = 0; j < 100; j++) {
					data[i][j] = Integer.parseInt(st.nextToken());
				}
			}
      
			int max = 0;

			for (int i = 0; i < 100; i++) {
				int sum = 0;
				for (int j = 0; j < 100; j++) {
					sum += data[i][j];
				}
				if (sum > max) {
					max = sum;
				}
			}

			for (int i = 0; i < 100; i++) {
				int sum = 0;
				for (int j = 0; j < 100; j++) {
					sum += data[j][i];
				}
				if (sum > max) {
					max = sum;
				}
			}

			int sum1 = 0;
			for (int i = 0; i < 100; i++) {
				sum1 += data[i][i];
			}
			if (sum1 > max) {
				max = sum1;
			}

			int sum2 = 0;
			for (int i = 0; i < 100; i++) {
				sum2 += data[i][99 - i];
			}
			if (sum2 > max) {
				max = sum2;
			}

			System.out.println("#" + t + " " + max);

		}
	}
}
